cmd/monaco/download/download_configs.go

```diff
@@ -145,10 +158,10 @@ func downloadConfigs(apis api.ApiMap, opts downloadOptions) (project.ConfigsPerT
 
 	c = client.LimitClientParallelRequests(c, opts.concurrentDownloadLimit)
 
-	apisToDownload, errors := getApisToDownload(apis, opts.specificAPIs)
-	if len(errors) > 0 {
-		err = PrintAndFormatErrors(errors, "failed to load apis")
-		return nil, err
+	apisToDownload := getApisToDownload(apis, opts.specificAPIs)
+	if len(apisToDownload) == 0 {
+		util.PrintError(fmt.Errorf("no APIs to download"))
```

Comment:
as you already fmt an error message here, I would also just return this error.
Also if the returned error from this method is printed further up the chain, there might not be a need for PrintError on this level at all.

Reply:
But it is more clear if result is just returned from the sub function, and than on the upper lever check result and make decision. 
It will also be easier for further refactoring

Reply:
this comment still stands - did you have a reason not to address it? :)

Reply:
furthermore, I would even go without error at this place.
That kind of error should happen before entering pkg command in validation phase of preparing data. and this should work fine if empty slice is passed.

Reply:
I don't quite understand your argument, maybe mine wasn't worded well either.
 
Yes this should absolutely return an error.
I meant instead of 

```
util.PrintError(fmt.Errorf("..."))
return fmt.Errorf("...")
```

do e.g. 
```
err := fmt.Errorf("...")
util.PrintError(err)
return fmt.Errorf(err)
```

or if the method that ends up calling this already prints the error when handling it (which it probably does, as your top-level error handling boils down to "tell the user"?) just do:
```
return fmt.Errorf(err)
```